cf/commands/domain/domains_test.go

```diff
@@ -1,137 +1,379 @@
+		Context("when list domans for org returns error", func() {
 			BeforeEach(func() {
-				domainRepo.ListDomainsForOrgStub = func(orgGuid string, cb func(models.DomainFields) bool) error {
-					cb(models.DomainFields{
-						Shared: false,
-						Name:   "Private-domain1",
-					})
+				domainRepo.ListDomainsForOrgReturns(errors.New("org-domain-err"))
+			})
 
-					cb(models.DomainFields{
-						Shared: true,
-						Name:   "The-shared-domain",
-					})
+			It("fails with message", func() {
+				Expect(func() { cmd.Execute(flagContext) }).To(Panic())
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"FAILED"},
+					[]string{"Failed fetching domains."},
+					[]string{"org-domain-err"},
+				))
+			})
+		})
 
-					cb(models.DomainFields{
-						Shared: false,
-						Name:   "Private-domain2",
-					})
+		Context("when there are no domains for org", func() {
+			BeforeEach(func() {
+				domainFields = []models.DomainFields{}
+				domainRepo.ListDomainsForOrgStub = callBackFunc
+				cmd.Execute(flagContext)
+			})
+
+			It("prints no domains found message", func() {
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"No domains found"},
+				))
+			})
 
-					return nil
+			It("does not print table header", func() {
+				Expect(ui.Outputs).ToNot(ContainSubstrings(
+					[]string{"name", "status", "type"},
+				))
+			})
+		})
+
+		Context("when domains are found", func() {
+			BeforeEach(func() {
+				domainFields = []models.DomainFields{
+					models.DomainFields{Shared: false, Name: "Private-domain1"},
+					models.DomainFields{Shared: true, Name: "Shared-domain1"},
 				}
+				domainRepo.ListDomainsForOrgStub = callBackFunc
 			})
 
-			It("lists domains", func() {
-				runCommand()
+			It("does not print no domains found message", func() {
+				cmd.Execute(flagContext)
+				Expect(ui.Outputs).NotTo(ContainSubstrings(
+					[]string{"No domains found"},
+				))
+			})
+
+			It("prints the table headers", func() {
+				cmd.Execute(flagContext)
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"name", "status", "type"},
+				))
+			})
 
-				orgGUID, _ := domainRepo.ListDomainsForOrgArgsForCall(0)
-				Expect(orgGUID).To(Equal("my-org-guid"))
+			It("prints the domain information", func() {
+				cmd.Execute(flagContext)
 				Expect(ui.Outputs).To(ContainSubstrings(
-					[]string{"Getting domains in org", "my-org", "my-user"},
-					[]string{"name", "status"},
-					[]string{"The-shared-domain", "shared"},
 					[]string{"Private-domain1", "owned"},
-					[]string{"Private-domain2", "owned"},
 				))
 			})
-		})
 
-		It("displays a message when no domains are found", func() {
-			runCommand()
-			Expect(ui.Outputs).To(ContainSubstrings(
-				[]string{"Getting domains in org", "my-org", "my-user"},
-				[]string{"No domains found"},
-			))
-		})
+			Context("when router api endpoint is not set", func() {
+				BeforeEach(func() {
+					factory.NewRoutingAPIRequirementReturns(
```

Comment:
This doesn't seem like it should be here. That it's a failing requirement isn't used by the tests below; it could just as well be a passingRequirement.